src/lang-go.ts

```diff
@@ -189,31 +190,22 @@ async function tryToCreateAccessToken(): Promise<string | undefined> {
 }
 
 async function connectAndInitialize(
-    address: string,
+    address: URL,
     root: URL,
     token: string | undefined
 ): Promise<rpc.MessageConnection> {
-    const connectingToGoLangserverHelp = [
-        `Unable to connect to the Go language server at ${address}.`,
-        `Make sure ${'go.address' as keyof Settings} in your Sourcegraph settings is set to the address of the language server (e.g. wss://sourcegraph.example.com/go).`,
-        `Read the full documentation for more information: https://github.com/sourcegraph/sourcegraph-go`,
-    ].join('\n')
-
-    const connectingToSourcegraphHelp = [
-        `The Go language server running on ${address} was unable to fetch repository contents from Sourcegraph running on ${sourcegraphURL()}.`,
-        `Make sure ${'go.sourcegraphUrl' as keyof Settings} in your settings is set to the address of Sourcegraph from the perspective of the language server (e.g. http://sourcegraph-frontend:30080 when running in Kubernetes).`,
-        `Read the full documentation for more information: https://github.com/sourcegraph/sourcegraph-go`,
-    ].join('\n')
-
-    const connection = (await new Promise((resolve, reject) => {
+    const connection = (await new Promise<wsrpc.MessageConnection>((resolve, reject) => {
         try {
-            const webSocket = new WebSocket(address)
+            const webSocket = new WebSocket(address.href)
             const conn = createWebSocketConnection(wsrpc.toSocket(webSocket), new ConsoleLogger())
             webSocket.addEventListener('open', () => resolve(conn))
-            webSocket.addEventListener('error', event => reject(new Error(connectingToGoLangserverHelp)))
+            webSocket.addEventListener('error', event => {
+                notifyUnableToConnectToLanguageServer(address)
+                reject(event)
+            })
         } catch (e) {
             if ('message' in e && /Failed to construct/.test(e.message)) {
```

Comment:
This `err.message` check (as any `err.message` check really) seems really brittle. What case is it trying to detect? Which case is it trying to not detect?

For example, I notice that it gets triggered in Chrome when you pass an `http://` URL. Firefox however has a different Error message for that case. It also does not get triggered for an unsuccessful connection, or for passing an invalid URL.